### PR TITLE
Geolocation Permissions Policy tests

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Associate spec changes: https://github.com/w3c/geolocation-api/pull/41 ... I wonder if any folks tagged would be kind enough to review this? I know geolocation is a bit old, but I’m hoping we can republish an updated W3C Rec that includes support for Feature Policy.

**Comment by engedy** [comment]
Thanks you for adding the tests, Marcos! I am afraid I don't have approval powers here, but LGTM!

**Comment by marcoscaceres** [comment]
@clelland, any chance you could review these?

File: geolocation-API/enabled-by-feature-policy-attribute.https.sub.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  "use strict";
+
+  const same_origin_src =
+    "/feature-policy/resources/feature-policy-geolocation.html";
+  const cross_origin_src =
+    "https://{{domains[www]}}:{{ports[https][0]}}" + same_origin_src;
+
+  async_test(t => {
+    test_feature_availability(
+      'geolocation',
```
Comment (reillyeon) [inline geolocation-API/enabled-by-feature-policy-attribute.https.sub.html:16]: Be consistent about " vs '.

```suggestion
      "geolocation",
```

File: geolocation-API/enabled-by-feature-policy-attribute.https.sub.html [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      t,
+      same_origin_src,
+      expect_feature_available_default,
+      "geolocation"
+    );
+  }, 'Feature policy "geolocation" can be enabled in same-origin iframe using allow="geolocation" attribute');
+
+  async_test(t => {
+    test_feature_availability(
+      'geolocation',
```
Comment (reillyeon) [inline geolocation-API/enabled-by-feature-policy-attribute.https.sub.html:26]:
```suggestion
      "geolocation",
```

File: geolocation-API/enabled-on-self-origin-by-feature-policy.https.sub.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        false
+      );
+      await new Promise(resolve => {
+        navigator.geolocation.getCurrentPosition(resolve);
+      })
+    }, 'Feature-Policy header geolocation "self" allows the top-level document.');
+
+    async_test((t) => {
+      test_feature_availability(
+        'geolocation',
```
Comment (reillyeon) [inline geolocation-API/enabled-on-self-origin-by-feature-policy.https.sub.html:30]:
```suggestion
        "geolocation",
```

File: geolocation-API/enabled-on-self-origin-by-feature-policy.https.sub.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        t,
+        same_origin_src,
+        expect_feature_available_default
+      );
+    }, 'Feature-Policy header geolocation "self" allows same-origin iframes.');
+
+    async_test((t) => {
+      test_feature_availability(
+        'geolocation',
```
Comment (reillyeon) [inline geolocation-API/enabled-on-self-origin-by-feature-policy.https.sub.html:39]:
```suggestion
        "geolocation",
```

**Review by reillyeon — APPROVED** [review]

**Pull request merged**